### PR TITLE
Add index to social embeds

### PR DESCRIPTION
**Pull request by jroebu14**
Partially resolves infra issue 1422

**Overall change:**
Adds indexes to social media embed blocks so that embed skip links contain the index/position of the embed type in optimo articles.

**Code changes:**

- Creates a new data transformer function called `addIndexesToEmbeds` that is added to the article route's data transformation stage.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: scripts/bundleSize/bundleSizeConfig.js [hunk 1]
```diff
@@ -4,6 +4,6 @@ module.exports = {
   // Keep the MAX_SIZE +5 above the largest value and MIN_SIZE -5
   // below the smallest value in the build output; this avoids the
   // need for frequent changes as bundle sizes fluctuate.
-  MIN_SIZE: 617,
-  MAX_SIZE: 898,
+  MIN_SIZE: 624,
+  MAX_SIZE: 909,
```
Comment (jroebu14) [inline scripts/bundleSize/bundleSizeConfig.js:8]: Bundle sizes before changes were made:

Screenshot 2021-07-03 at 15 26 59

Bundle sizes after changes:

Screenshot 2021-07-03 at 15 29 42

File: src/app/containers/SocialEmbed/index.jsx [hunk 2]
```diff
@@ -47,7 +46,7 @@ const SocialEmbedContainer = ({ blocks, source }) => {
 
   const skipLink = {
     ...skipLinkTranslations,
-    endTextId: `skip-%provider%-content-${index}`,
+    endTextId: `skip-%provider%-content-${oEmbedPosition}`,
```
Comment (greenc05) [inline src/app/containers/SocialEmbed/index.jsx]: Not sure if this is just an example? Anchor link text should ideally be named as per the destination e.g. "end-of-twitter-content-1"

Comment (jroebu14) [reply src/app/containers/SocialEmbed/index.jsx]: That's just the id on the element. The text is still correct. Does this look ok to you?

Screenshot 2021-07-13 at 09 36 39

Comment (greenc05) [reply src/app/containers/SocialEmbed/index.jsx]: The id forms part of the url, it should ideally be named as per the destination

Comment (jroebu14) [reply src/app/containers/SocialEmbed/index.jsx]: ok I will fix this

**Review by HarryVerhoef — APPROVED** [review]
Nice one! 🚀

**Review by andrewscfc — APPROVED** [review]
Nice work, really comprehensive tests for some pretty complex logic; they get across the intent very clearly

**Pull request merged**